search/src/Components/upload/upload_stepper.js

```diff
@@ -206,42 +27,80 @@ const useStyles = (theme) => ({
   },
 });
 
-function getSteps() {
-  return ['Upload Weedcoco', 'Add Agcontext', 'Upload Images'];
+function getSteps(upload_type) {
+  return upload_type === 'coco'?
+         ['Upload Coco', 'Add Agcontext', 'Upload Images']:
+         upload_type === 'weedcoco'?
+         ['Upload Weedcoco', 'Upload Images']:
+         ['Upload Coco', 'Add Agcontext', 'Upload Images']
 }
 
-function getStepContent(step, upload_id, images, handleUploadId, handleImages) {
-  switch (step) {
-    case 0:
-      return <UploaderSingle upload_id={upload_id} images={images} handleUploadId={handleUploadId} handleImages={handleImages}/>;
-    case 1:
-      return 'Placeholder for a Agcontext form';
-    case 2:
-      return <UploaderImages upload_id={upload_id} images={images}/>;
-    default:
-      return 'Unknown step';
-  }
+function getStepContent(step, upload_type, upload_id, images, formData, handleUploadId, handleImages, handleFormData, handleUploadAgcontexts) {
+    if (upload_type === 'coco') {
+        switch (step) {
+            case 0:
+              return <UploaderSingle upload_id={upload_id} images={images} handleUploadId={handleUploadId} handleImages={handleImages}/>;
+            case 1:
+              return (
+                <React.Fragment>
+                    <AgContextForm formData={formData} onChange={e => handleFormData(e.formData)} />
+                    <textarea style={{width: "100%", height: "5em"}} value={toJSON(formData)} ></textarea>
```

Comment:
(Unless there's also a way to post the agcontext in this format, there's not much value in being able to export the JSON. This can be fixed, e.g. allowing the user to paste in new content and the form to update, in another PR.)